contrib/cr3bp_DhruvJ/cr3bp_char_quant.py

```diff
@@ -0,0 +1,156 @@
+        dhruvj9922@gmail.com
+
+Objective: This file contains functions to compute various characteristic quantities
+           for a Cirular Restircted Three Body Problem (CR3BP) model setup
+           
+           Features: 
+               1. Compute CR3BP 'mu', characterisitic mass of P1-P2 primary bodies
+               2. Compute CR3BP 'l*', characterisitic lenght of P1-P2 primary bodies
+               3. Comptue CR3BP 't*', charcterisitic time of P1-P2 primary bodies
+               
+Useful ideas: 
+1. To non-dimensioanlize a physical position vector: divide by l*
+2. To dimensioanlize a [nd] position vector: multiply by l*
+3. To non-dimensionalize a physical veloctiy vecotr, multiply by t*/l*
+4. To dimensionalize a [nd] veloctiy vecotr, multiply by l*/t*
+5. Position of P1 = [-mu, 0, 0] [nd] {x,y,z}
+5. Position of P2 = [1-mu, 0, 0] [nd] {x,y,z} 
+
+Physical Qunatities obtained from: JPL’s ephemerides file de405.spk and https://ssd.jpl.nasa.gov/?planet_pos
+"""
+import copy
+
+import numpy as np
+
+
+def mu_calc(mu_pi):
+    """Calculate mu of CR3BP
+    Parameters
+    ----------
+    mu_pi : ndarray, float
+        mu_pi[0] = mu of P1
+        mu_pi[1] = mu of P2
+
+    Returns
+    -------
+    mu: float, M2/(M1+M2)
+        M1 and M2 are mass of Primary Bodies and M2<M1
+    """
+    return mu_pi[1] / (mu_pi[0] + mu_pi[1])
+
+
+def tstar_calc(mu_pi, dist):
+    """Calculate t* of CR3BP
+    Parameters
+    ----------
+    dist: float, [nd]
+        Non-dimensional distance between P1 and P2
+    mu_pi: ndarray, float
+        mu_pi[0] = mu of P1
+        mu_pi[1] = mu of P2
+
+    Returns
+    -------
+    t*: float, [nd]
+        sqrt(dist^3/m*)
+        Non-dimensional time of P1-P2 system
+    """
+    return (dist**3 / (mu_pi[0] + mu_pi[1])) ** 0.5
+
+
+def bodies_char(b1, b2):
+    """Calculates mu, dist[km], t* [nd] of the 'body_i - body_j system'
+    Parameters
+    ----------
+    b1 : string
+       'body_i'
+    b2 : string
+       'body_j'
+
+    Returns
+    -------
+    mu: float
+        mu2/(mu1+mu2), mu2<mu1
+    dist: float, [nd]
+        Non-dimensional distance between P1 and P2
+    tstar: float, [nd]
+        sqrt(dist^3/m*)
+        Non-dimensional time of P1-P2 system
+    """
+    # Body values
+    mu = {}  # km^4 kg^-1 s^-2
+    mu["Sun"] = 132712440017.99
+    mu["Moon"] = 4902.8005821478
+    mu["Earth"] = 398600.4415
+
+    mu["Mars"] = 42828.314258067  # Mars, GM
+    mu["Jupiter"] = 126712767.8578  # Jupiter, GM
+    mu["Saturn"] = 37940626.061137  # Saturn, GM
+    mu["Uranus"] = 5794549.0070719  # Uranus, GM
+    mu["Neptune"] = 6836534.0638793  # Neptune, GM
+    mu["Pluto"] = 981.600887707  # Pluto, GM
+
+    mu["Phobos"] = 0.0007112  # Phobos, GM
+    mu["Titan"] = 8978.1382  # Titan, GM
+    mu["Ganymede"] = 9887.834  # Ganymede, GM
+    mu["Titania"] = 228.2  # Titania, GM
+    mu["Triton"] = 1427.598  # Triton, GM
+    mu["Charon"] = 102.30  # Charon, GM
```

Comment:
Should this use `poliastro.constants` instead?

Reply:
yes, I am looking into leveraging that function to more seamlessly incorporate this code to poliastro/src in the future.